dbt/adapters/athena/impl.py

```diff
@@ -688,4 +696,7 @@ def get_columns_in_relation(self, relation: AthenaRelation) -> List[Column]:
 
         logger.debug(f"Columns in relation {relation.identifier}: {columns + partition_keys}")
 
-        return [Column(c["Name"], c["Type"]) for c in columns + partition_keys]
+        return [
+            AthenaColumn(column=c["Name"], dtype=c["Type"], is_iceberg=self._is_iceberg_table(table))
```

Comment:
why instead of passing `is_iceberg ` don't you pass something more flexible like table type? This could be more future proof when more "table" types will be supported in the future.